src/persist-types/src/lib.rs

```diff
@@ -95,7 +95,7 @@ pub mod part;
 
 /// Encoding and decoding operations for a type usable as a persisted key or
 /// value.
-pub trait Codec: Sized + 'static {
+pub trait Codec: Send + Sync + Sized + 'static {
```

Comment:
needed for BatchFetcher to be moved into a task. not sure how controversial this is, but it seemed preferable to threading through `+ Send + Sync` to `K` and `V` everywhere

Reply:
This seems totally reasonable to me!

Reply:
historically, whenever this has cropped up, it's meant there's a PhantomData somewhere with an incorrect variance: https://doc.rust-lang.org/beta/nomicon/phantom-data.html#table-of-phantomdata-patterns

Reply:
Since I had a [less capable version of this on a recent branch](https://github.com/MaterializeInc/materialize/pull/17855/commits/b61b3e27829bb7177d3e73f6ae42e38a57480257), I had a look to see why I didn't seem to need this... and IIUC it's because I only moved the _reference_ into the task, which is fine since the relevant methods only take `&self`. Maybe an option here as well.

Reply:
@bkirwi `FuturesUnordered` is single-threaded!

Reply:
Hmm, yeah, in retrospect almost certainly has more to do with the fact that I wasn't spinning up an additional task at all (which I works fine for this case?) than any particularity of the async block.